tests/framework/extensions/provisioning/creates.go

```diff
@@ -227,22 +227,33 @@ func CreateProvisioningCustomCluster(client *rancher.Client, externalNodeProvide
 		}
 	}
 
-	if clustersConfig.Hardened && clustersConfig.KubernetesVersion <= string(provisioninginput.HardenedKubeVersion) {
-		err = hardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode)
-		if err != nil {
-			return nil, err
-		}
+	if clustersConfig.Hardened {
+		if strings.Contains(clustersConfig.KubernetesVersion, clusters.K3SClusterType.String()) {
+			err = k3sHardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode, clustersConfig.KubernetesVersion)
```

Comment:
Just to make sure, if the cluster is a k3s cluster we need add the k8s version? What's the reason? Why doesn't RKE2 need the same

Reply:
Because when you have K8s v1.24 and below, the hardening process changes exclusively for K3S; RKE2 stays the same regardless of the version.